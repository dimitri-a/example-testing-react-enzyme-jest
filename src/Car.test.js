import React from 'react';
import { shallow } from 'enzyme';
import { Car } from './Car';
import Engine from './Engine';

describe('<Car />', () => {
  let wrapper;
  let handleStartCar;
  beforeAll(() => {
    handleStartCar = jest.fn();
    wrapper = shallow(<Car color='red' handleStartCar={handleStartCar} />);
  });
  // check dynamic data renders correctly
  // find elements by ID
  it('renders the heading', () => {
    const heading = wrapper.find('#car--heading');
    expect(heading).toHaveLength(1);
    expect(heading.text()).toEqual('I am a red car');
  });
  // check child components are called correctly
  it('has a diesel engine', () => {
    // testing hint: If the child component is a composed component
    // then you will likely need to find by class otherwise
    // finding by string (e.g. 'Engine') will work
    const engine = wrapper.find(Engine);
    expect(engine.prop('type')).toEqual('diesel');
  });
  // check actions
  it('calls hanldeStartCar on button click', function () {
    const button = wrapper.find('#car--start');
    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(handleStartCar).toHaveBeenCalledTimes(1);
  });
});
