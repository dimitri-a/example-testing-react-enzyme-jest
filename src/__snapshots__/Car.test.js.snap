// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Car /> matches snapshot 1`] = `
<div>
  <h1
    id="car--heading"
    style={
      Object {
        "color": "red",
      }
    }
  >
    I am a 
    red
     car
  </h1>
  <Engine
    type="diesel"
  />
  <button
    id="car--start"
    onClick={[Function]}
    type="button"
  >
    Start
  </button>
</div>
`;
